testkit/src/mill/testkit/IntegrationTesterBase.scala

```diff
@@ -77,7 +77,9 @@ trait IntegrationTesterBase {
       } else {
         val serverPath0 = outDir / millNoDaemon
 
-        for (serverPath <- os.list.stream(serverPath0)) os.remove(serverPath / processId)
+        if (os.exists(serverPath0))
```

Comment:
Is there a way to avoid this conditional? Adding it makes this code very fragile, since if the server path changes or is misconfigured the code will fail silently and cause process leaks in our integration test suite causing other random failures

Reply:
I think I added that because I must have run in this conditional not to be true at some point when working on this. I'm going to revert it, and see what happens.